pkg/operator/starter.go

```diff
@@ -115,33 +158,64 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 			}
 			return isSNO
 		},
-	).AddKubeInformers(kubeInformersForNamespaces)
+	).AddKubeInformers(managementKubeInformersForNamespaces)
 
-	controllerDeploymentManifest, err := assets.ReadFile("csi_controller_deployment.yaml")
+	controllerDeploymentManifest, err := namespacedAssetFunc("csi_controller_deployment.yaml")
 	if err != nil {
 		return err
 	}
+
+	var webhookHooks []validatingWebhookConfigHook
+	if isHyperShift {
+		webhookHooks = []validatingWebhookConfigHook{
+			hyperShiftSetWebhookService(),
+		}
+	}
+	webhookController := NewValidatingWebhookConfigController(
+		"WebhookController",
+		guestOperatorClient,
+		guestKubeClient,
+		guestKubeInformersForNamespaces.InformersFor("").Admissionregistration().V1().ValidatingWebhookConfigurations(),
+		namespacedAssetFunc,
+		"webhook_config.yaml",
+		controllerConfig.EventRecorder,
+		webhookHooks,
+	)
+
+	var deploymentHooks []dc.DeploymentHookFunc
+	if isHyperShift {
+		deploymentHooks = []dc.DeploymentHookFunc{
+			hyperShiftReplaceNamespaceHook(operandNamespace),
+			hyperShiftRemoveNodeSelector(),
+			hyperShiftAddKubeConfigVolume("admin-kubeconfig"), // TODO: use dedicated secret for Snapshots
```

Comment:
I was planning to always pass the same hooks (hypershift or not) and then add some logic inside the hook to skip if the cluster if it's not hypershift: https://github.com/openshift/aws-ebs-csi-driver-operator/pull/159/files#diff-0d623dfd885adb20f991bda4c2453aebd732ca6dbb4d1d4be6e79805c3b48de6R460

Which is better?

Reply:
I had common hooks initially, but then there were too many branches. E.g. selector should be different, tolerations, replicas, kubeconfig volumes, ...